clients/cobol-lsp-vscode-extension/src/commands/RunAnalysisCLI.ts

```diff
@@ -0,0 +1,259 @@
+
+    if (extensionFolder && currentFileLocation !== "") {
+      return (
+        'java -jar "' +
+        extensionFolder +
+        '" ' +
+        this.buildAnalysisCommandPortion(currentFileLocation)
+      );
+    }
+
+    return "";
+  }
+
+  /**
+   * Provides the portion of the command from "analysis" onwards.
+   * Is the same for both the Java and Native builds.
+   * @param currentFileLocation - Location of the main cobol file being analyzed.
+   * @protected
+   */
+  protected buildAnalysisCommandPortion(currentFileLocation: string) {
+    const copyBookCommand = `-cf=${
+      this.copybookConfigLocation === ""
+        ? "."
+        : '"' + this.copybookConfigLocation + '"'
+    }`;
+
+    return 'analysis -s "' + currentFileLocation + '" ' + copyBookCommand;
+  }
+
+  /**
+   * Sends a given command to a terminal.
+   * Checks to see if one named "Analysis" is already created, if so clear and reuse it.
+   * @param command - The command to run from the terminal.
+   * @protected
+   */
+  protected sendToTerminal(command: string) {
+    const existingTerminal = vscode.window.terminals.find(
+      (term: Terminal) => term.name === "Analysis",
+    );
+    const terminal = existingTerminal
+      ? existingTerminal
+      : vscode.window.createTerminal("Analysis");
+
+    if (existingTerminal) {
+      terminal.sendText(this.getClearCommand(process.platform));
+    }
+
+    terminal.sendText(command);
+    terminal.show(true);
+  }
+
+  /**
+   * Retrieves the location of the currently open file.
+   * If it is not saved, creates a temporary one on the disk to send to the command line with a copy of the contents.
+   *    - Does not result in the actual file being saved to the end user.
+   * @protected
+   */
+  protected getCurrentFileLocation() {
+    if (vscode.window.activeTextEditor) {
+      if (
+        vscode.window.activeTextEditor.document.uri.path
+          .toString()
+          .startsWith("Untitled")
+      ) {
+        return this.saveTempFile();
+      } else {
+        return vscode.window.activeTextEditor.document.uri.path;
+      }
+    }
+    return "";
+  }
+
+  /**
+   * Creates a temporary one on the disk to send to the command line with a copy of the contents.
+   *    - Does not result in the actual file being saved to the end user.
+   * @protected
+   */
+  protected saveTempFile() {
+    const data = vscode.window.activeTextEditor?.document.getText();
+
+    if (data) {
+      try {
+        vscode.workspace.fs
+          .stat(this.globalStorageUri)
+          .then((fileStatResult) => {
+            if (fileStatResult.type === vscode.FileType.Directory) {
+              const newFileName = Date.now() + ".cbl";
+              const newUri = vscode.Uri.parse(
+                this.globalStorageUri.path + "/" + newFileName,
+              );
+              vscode.workspace.fs.writeFile(newUri, Buffer.from(data));
+              return newUri.path;
```

Comment:
This is actually never returned from the `saveTempFile`.

Reply:
It does. I've tested it with temporary files without issues.

Reply:
I've tried to analyze an untitled document and nothing happens.

Reply:
The temp file function has been rewritten, please try again. Now avoids using lambdas within `then()` chains and has proper `await/async` functionality.